dspy/backends/template.py

```diff
@@ -12,12 +13,16 @@ class TemplateBackend(BaseBackend):
 
     lm: BaseLM
 
-    def __call__(
+    @property
+    def lm_history(self):
+        return self.lm.history
+
```

Comment:
Seems a little extra

Reply:
I agree. I was using it for testing. Will likely remove all lm based history tracking and just keep it at the backend level, unless we find we need it somewhere else.